kong/pdk/request.lua

```diff
@@ -625,6 +626,85 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the value of the specified request cookie.
+  --
+  -- The returned value is either a `string`, or can be `nil` if a cookie with
+  -- `name` was not found in the request.
+  --
+  -- Cookie names are case-insensitive.
+  --
+  -- @function kong.request.get_cookie
+  -- @phases rewrite, access, header_filter, body_filter, log, admin_api
+  -- @tparam string name the name of the cookie to be returned
+  -- @treturn string|nil the value of the cookie or nil if not present
+  -- @usage
+  -- -- Given a request with the following cookies:
+  --
+  -- -- Cookie: X-Cookie-Foo=Hello; X-Cookie-Bar=World
+  --
+  -- kong.request.get_cookie("X-Cookie-Foo")        -- "Hello"
+  -- kong.request.get_cookie("X-Cookie-Bar")        -- "World"
+  -- kong.request.get_cookie("X-Cookie-foo")        -- "Hello"
+  -- kong.request.get_cookie("X-Cookie-Missing")    --  nil
+  function _REQUEST.get_cookie(name)
+    check_phase(PHASES.request)
+
+    if type(name) ~= "string" then
+      error("cookie name must be a string", 2)
+    end
+
+    return ngx.var["cookie_" .. name]
+  end
+
+
+  local function cookie_iterator_to_tab(iterator)
+    local cookie_tab = {}
+    while true do
+        local match = iterator()
+        if not match then
+            break
+        end
+        cookie_tab[pl_stringx.strip(match[1],' ')] = match[2]
```

Comment:
this block uses 4 spaces indent, instead of 2

Also, since this is a very simple function I'd probably inline it in the function below. If you do not inline it, then at least wrap it in a do-end block to localize it to the function below.